python/pyarrow/table.pxi

```diff
@@ -2926,6 +2926,20 @@ cdef class RecordBatch(_Tabular):
                 <CResult[shared_ptr[CArray]]>deref(c_record_batch).ToStructArray())
         return pyarrow_wrap_array(c_array)
 
+    def to_tensor(self):
+        """
+        Convert to a :class:`~pyarrow.Tensor`.
```

Comment:
For a next PR, you can document a bit more in detail the constraints of when this will work